sympy/geometry/point.py

```diff
@@ -127,93 +149,40 @@ def __new__(cls, *args, **kwargs):
             return Point2D(*coords, **kwargs)
         elif len(coords) == 3:
-            kwargs['_ok'] = True
+            kwargs['_nocheck'] = True
             return Point3D(*coords, **kwargs)
-        elif len(coords) < 2:
-            raise ValueError('Point requires 2 or more coordinates or a dimension > 1.')
 
         # the general Point
         return GeometryEntity.__new__(cls, *coords)
 
     is_Point = True
 
-    @property
-    def _size(p):
-        # return the location of the highest nonzero dimension
-        # or 2, whichever is greater
-        return max([1] + [i for i,x in enumerate(p) if x != 0]) + 1
-
-    @staticmethod
-    def normalize_dimensions(*points, **kwargs):
-        """Return all Points with the same dimensions.
-
-        Set the keyword `dim` to the desired number of dimensions.
-        If it is not given then the smallest dimension necessary to
-        capture all non-zero coordinates will be used; but if the
-        keyword `max` is True then all dimensions will match the
-        largest dimension present.
-
-        Raises
-        ======
-
-        ValueError : no points are given or when attempting to
-            truncate a nonzero dimension
-        TypeError : an element was not a Point instance
-
-        Examples
-        ========
-
-        >>> from sympy.geometry import Point
-
-        By default, extra dimensions with zero elements are trimmed:
-
-        >>> Point.normalize_dimensions(Point(1, 2, 0), Point(1, 2, 3, 0))
-        [Point3D(1, 2, 0), Point3D(1, 2, 3)]
-
-        Zeros will be padded onto Points that are too short:
-
-        >>> a, b = Point(1, 2), Point(1, 2, 3)
-        >>> Point.normalize_dimensions(a, b)
-        [Point3D(1, 2, 0), Point3D(1, 2, 3)]
-
-        The keyword `dim` will force a given dimension:
-
-        >>> Point.normalize_dimensions(a, b, dim=4)
-        [Point(1, 2, 0, 0), Point(1, 2, 3, 0)]
-
-        But if truncation would result in losing a non-zero dimension,
-        a ValueError is raised:
-
-        >>> Point.normalize_dimensions(a, b, dim=2)
-        Traceback (most recent call last):
-        ...
-        ValueError: Point has too many dimensions
-
-        To raise a ValueError whenever dimensions are being lost
-        set the keyword `warn` to True.
-        """
-        if not points:
-            raise ValueError('Must provide 1 or more Points.')
-        if not all(isinstance(i, Point) for i in points):
-            raise TypeError('all elements must be Points')
-        minimal = not kwargs.get('max', False)
-        dim = kwargs.get('dim',
-            max(i._size for i in points) if minimal else
-            max(i.ambient_dimension for i in points))
-        warn = kwargs.get('warn', False)
-        if not all(a.ambient_dimension == dim for a in points):
-            points = [Point(a, dim=dim, warn=warn) for a in points]
-        return points
+    @classmethod
+    def _normalize_dimension(cls, *args, **kwargs):
```

Comment:
I preferred to require that the args be Points since the calling routine knows better what is and isn't a Point. The way it is now, if dim is not given then two additional Point calls are being made. It was my impression that Point is rather slow. Perhaps with the mods the speed for handling Point instances is better and this is no longer a significant issue?